### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by PyCore23**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1 +1,11 @@
-# write your code here
+from os import path, listdir, getcwd, chdir
+from sys import argv
+
+
+def copy_file(command: str) -> None:
+    chdir(path.dirname(argv[0]))
+    ref = command.split(" ")[1]
```
Comment (vasylhnatiuk) [inline app/main.py:7]: what is ref? what if command does not have 3 arguments?

File: app/main.py [hunk 2]
```diff
@@ -1 +1,11 @@
...
+    new = command.split(" ")[2]
```
Comment (vasylhnatiuk) [inline app/main.py:8]: use only one time split(), don't repeat yourself

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 3]
```diff
@@ -1 +1,17 @@
-# write your code here
+from os import path, chdir
+from sys import argv
+
+
+def copy_file(command: str) -> None:
+    parts = command.split(" ")
```
Comment (EdAlekseiev) [inline app/main.py:6]: `" "` is default value for `split`.
```suggestion
    parts = command.split()
```

File: app/main.py [hunk 4]
```diff
@@ -1 +1,17 @@
...
+    if len(parts) < 3:
```
Comment (EdAlekseiev) [inline app/main.py:7]: Also, check that command start with `cp`

**Review by EdAlekseiev — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 5]
```diff
@@ -1 +1,13 @@
-# write your code here
+def copy_file(command: str) -> None:
+    parts = command.split(" ")
```
Comment (vsmutok) [inline app/main.py:2]:
```suggestion
    parts = command.split()
```

Comment (vsmutok) [reply app/main.py:2]: `" "` is default value for `split()`

File: app/main.py [hunk 6]
```diff
@@ -1 +1,13 @@
...
+    if len(parts) < 3:
```
Comment (vsmutok) [inline app/main.py:3]: Also, check that command start with `cp`

File: app/main.py [hunk 7]
```diff
@@ -1 +1,13 @@
...
+        print("invalid command format")
+    else:
+        source = parts[1]
+        new = parts[2]
```
Comment (vsmutok) [inline app/main.py:7]:
```suggestion
        _, source, new = parts
```

**Review by vsmutok — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 3 above)
Comment (vsmutok) [reply app/main.py:2]: `" "` is default value for `split()`

File: app/main.py [hunk 8]
```diff
@@ -1 +1,13 @@
-# write your code here
+def copy_file(command: str) -> None:
+    parts = command.split(" ")
+    if len(parts) < 3 or parts[1] == parts[2]:
+        raise ValueError("Invalid command")
+
+    try:
+        with open(parts[1], "r") as source_file:
+            with open(parts[2], "w") as destination_file:
```
Comment (vsmutok) [inline app/main.py:8]: use one `with`

```python
with open(.., "r") as f, open(..., "w") as f2:
```

**Review by vsmutok — CHANGES_REQUESTED** [review]

**Pull request closed without merging**